kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsFileServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+import edu.arizona.kfs.vnd.service.EcustomsFileService;
+
+/**
+ * UAF-66 MOD-PA7000-02 ECustoms - US Export Compliance
+ *
+ * New service to handle the creation of the eCustoms data file for validating vendor compliance with US Export Controls.
+ *
+ * @author Adam Kost kosta@email.arizona.edu
+ */
+public class EcustomsFileServiceImpl implements EcustomsFileService {
+
+    private static final String ERROR_MISSING_OUTPUT_FILE_DIRECTORY = "The server property ecustoms.output.directory is blank.";
+    private static final String ERROR_MISSING_OUTPUT_FILE_PREFIX = "The server property ecustoms.output.file.prefix is blank.";
+    private static final String ERROR_CANNOT_CREATE_DIRECTORY = "Unable to create directory structure.";
+
+    private static final Logger LOG = Logger.getLogger(EcustomsFileServiceImpl.class);
+    private String outputFileDirectory;
+    private String outputFileNamePrefix;
+
+    // private getter for error checking purposes, so that this error checking can be reused whenever this field is used.
+    private String getOutputFileDirectory() throws IOException {
+        if (outputFileDirectory == null || StringUtils.isBlank(outputFileDirectory)) {
+            throw new IOException(ERROR_MISSING_OUTPUT_FILE_DIRECTORY);
+        }
+        return outputFileDirectory;
+    }
+
+    // private getter for error checking purposes, so that this error checking can be reused whenever this field is used.
+    private String getOutputFileNamePrefix() throws IOException {
+        if (outputFileNamePrefix == null || StringUtils.isBlank(outputFileNamePrefix)) {
+            throw new IOException(ERROR_MISSING_OUTPUT_FILE_PREFIX);
+        }
+        return outputFileNamePrefix;
+    }
+
+    public void setOutputFileDirectory(String outputFileDirectory) {
+        this.outputFileDirectory = outputFileDirectory;
+        LOG.debug("outputFileDirectory=" + outputFileDirectory);
+    }
+
+    public void setOutputFileNamePrefix(String outputFileNamePrefix) {
+        this.outputFileNamePrefix = outputFileNamePrefix;
+        LOG.debug("outputFileNamePrefix=" + outputFileNamePrefix);
+    }
+
+    @Override
+    public File getDailyBatchDataFile(Date jobRunDate) throws IOException {
+        String filename = getDailyBatchDataFileName(jobRunDate);
+        File file = createFile(filename);
+        return file;
+    }
+
+    @Override
+    public File getDailyBatchVendorCountFile(Date jobRunDate) throws IOException {
+        String filename = getDailyBatchVendorCountFileName(jobRunDate);
+        File file = createFile(filename);
+        return file;
+    }
+
+    @Override
+    public void createDailyBatchDoneFile(Date jobRunDate) throws IOException {
+        String filename = getDailyBatchDoneFileName(jobRunDate);
+        File file = createFile(filename);
+        FileOutputStream fileOutputStream = new FileOutputStream(file);
+        fileOutputStream.write(VendorConstants.ECUSTOMS_DONE_MESSAGE_DAILY.getBytes());

```

Comment:
This write() needs to be in a try/catch w/ a finally clause that ensures the resource is closed. I've taken down JVM's with that mistake.